src/python/dxpy/workflow_builder.py

```diff
@@ -569,16 +622,32 @@ def skip_update():
     return global_workflow_id
 
 
-def _build_or_update_workflow(json_spec, args):
+def _build_or_update_workflow(args, parser):
     """
     Creates or updates a workflow on the platform.
     Returns the workflow ID, or None if the workflow cannot be created.
     """
     try:
         if args.mode == 'workflow':
+            json_spec = _parse_executable_spec(args.src_dir, "dxworkflow.json", parser)
             json_spec = _get_validated_json(json_spec, args)
             workflow_id = _build_regular_workflow(json_spec, args.keep_open)
         elif args.mode == 'globalworkflow':
+            if args._from:
+                json_spec = _fetch_spec_from_workflow(args, parser)
+            else:
+                json_spec = _parse_executable_spec(args.src_dir, "dxworkflow.json", parser)
+            
+            # Check if the local or source workflow is compiled by dxCompiler that supported dependency annotation
+            if json_spec.get("tag") and "dxCompiler" in json_spec["tag"]:
+                _check_dxcompiler_version(json_spec)
+                if not args.brief:
+                    _notify_instance_type_selection(json_spec)
+                    _notify_dependencies(json_spec)
```

Comment:
Here checking instance type selection and unbundled dependency only run when --brief is false, because it is just sending notifications to the user without changing the workflow.